nvm.sh

```diff
@@ -275,6 +275,357 @@ nvm_find_up() {
   nvm_echo "${path_}"
 }
 
+nvm_string_contains_regexp() {
+  local string
+  string="${1-}"
+  local regexp
+  regexp="${2-}"
+  if [ -z  "${string-}" ] || [ -z "${regexp-}" ]; then
+    return 1
+  fi
+  # e.g. "nvm_string_contains_regexp abbc ^aa?b+.$" returns 0
+  command printf "%s" "$string" | command awk "/$regexp/{ exit 0 }{ exit 1 }"
+}
+
+# scoped function returns nothing or a normalized semver adhering to the following grammar:
+#
+# semver         ::= comparator_set ( ' || '  comparator_set )*
+# comparator_set ::= comparator ( ' ' comparator )*
+# comparator     ::= ( '<' | '<=' | '>' | '>=' | '' ) [0-9]+ '.' [0-9]+ '.' [0-9]+
+nvm_validate_semver() { (
+  semver=$(command printf "%s" "${1-}" | command tr '||' '\n')
+  [ -n "$semver" ] || return 1
+  validated_semver='';
+  while [ -n "$semver" ]; do
+    comparator_set=$(command printf "%s" "$semver" | head -n1)
+    semver=$(command printf "%s" "$semver" | tail -n +2)
+    [ -n "$comparator_set" ] || continue
+
+    # convert comparators into required grammar
+    validated_comparator_set=$(command printf " %s " "$comparator_set" |
+      # exactly 1 space is needed before and after every comparator (including the first and last comparators)
+      command sed -E 's/\011/ /g;s/ +/ /g' |
```

Comment:
can all these consecutive sed commands be consolidated into one with multiple -E patterns?

Reply:
They could. But I was thinking it's more beneficial for debugging to have each sed replacement on it's own line with comments showing exactly what each is replacing instead of having one giant sed command. What do you think?

Reply:
I think you can do that as one command, see https://github.com/creationix/nvm/blob/master/nvm.sh#L1069-L1077 for an example? (iow, i think you should definitely keep each `-E` separate, with a comment, but calling `sed` multiple times doesn't seem to add much?